docs/guides/plugins.md

```diff
@@ -1,59 +1,344 @@
+
+### Setting up a Basic Plugin
+
+Finally, we can use our plugin on a player. There are two ways to do this. The first way is during creation of a Video.js player. Using the `plugins` option, a plugin can be automatically set up on a player:
 
 ```js
-    videojs.plugin('examplePlugin', examplePlugin);
+videojs('example-player', {
+  plugins: {
+    examplePlugin: {
+      customClass: 'example-class'
+    }
+  }
+});
 ```
 
-From this point on, your plugin will be added to the Video.js prototype and will show up as a property on every instance created. Make sure you choose a unique name that doesn't clash with any of the properties already in Video.js. Which leads us to...
+Otherwise, a plugin can be manually set up:
+
+```js
+var player = videojs('example-player');
+player.examplePlugin({customClass: 'example-class'});
+```
+
+These two methods are functionally identical - use whichever you prefer! That's all there is to it for basic plugins.
+
+## Writing a Class-Based Plugin
+
+As of Video.js 6, there is an additional type of plugin supported: class-based plugins.
+
+At any time, you may want to refer to the [Plugin API docs][api-plugin] for more detail.
 
-Step 3: Using A Plugin
-----------------------
-There are two ways to initialize a plugin. If you're creating your video tag dynamically, you can specify the plugins you'd like to initialize with it and any options you want to pass to them:
+### Write a JavaScript Class/Constructor
+
+If you're familiar with creating [components](components.md), this process is similar. A class-based plugin starts with a JavaScript class (a.k.a. a constructor function).
+
+This can be achieved with ES6 classes:
 
 ```js
-    videojs('vidId', {
-      plugins: {
-        examplePlugin: {
-          exampleOption: true
-        }
-      }
+const Plugin = videojs.getPlugin('Plugin');
+
+class ExamplePlugin extends Plugin {
+
+  constructor(player, options) {
+    super(player);
+
+    player.addClass(options.customClass);
+    player.on('playing', function() {
+      videojs.log('playback began!');
     });
+  }
+}
+```
+
+Or with ES5:
+
+```js
+var Plugin = videojs.getPlugin('Plugin');
+
+var ExamplePlugin = videojs.extend(Plugin, {
+
+  constructor: function(player, options) {
+    Plugin.prototype.call(this, player, options);
+
+    player.addClass(options.customClass);
+    player.on('playing', function() {
+      videojs.log('playback began!');
+    });
+  }
+});
+```
+
+For now, this example class-based plugin does the exact same thing as the basic plugin described above - not to worry, we will make it more interesting as we continue!
+
+### Register a Class-Based Plugin
+
+The registration process for class-based plugins is identical to [the process for basic plugins](#register-a-basic-plugin).
+
+```js
+videojs.registerPlugin('examplePlugin', ExamplePlugin);
+```
+
+### Setting up a Class-Based Plugin
```

Comment:
I think that makes sense. I'll reshuffle this doc a bit to eliminate some of the duplication.